src/System.IO.FileSystem/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -13,11 +13,6 @@ public sealed partial class SafeFileHandle : SafeHandle
     {
         public SafeFileHandle(IntPtr preexistingHandle, bool ownsHandle) : this(ownsHandle)
         {
-            long longHandle = (long)preexistingHandle;
```

Comment:
Can you help me understand this change?  I thought that since fd's were only 32bits we wanted to do this filtering on Unix.  Now that's not the case?  If have a safe file handle around Int32.MaxValue+1 or something, what happens?